app/workers/dequeue_subject_queue_worker.rb

```diff
@@ -2,19 +2,17 @@ class DequeueSubjectQueueWorker
   include Sidekiq::Worker
 
   sidekiq_options queue: :data_medium
+  sidekiq_options congestion: {
+      interval: 30,
+      max_in_interval: 1,
+      min_delay: 0,
+      reject_with: :cancel
+    }.merge({key: -> (queue_id, sms_ids) { "queue_#{ queue_id }_dequeue" }})
 
-  sidekiq_options congestion: { interval: 30, max_in_interval: 1, min_delay: 0, reject_with: :cancel }.merge({
-    key: ->(workflow_id, sms_ids, user_id, subject_set_id) {
-      "user_#{ workflow_id }_#{user_id}_#{subject_set_id}_subject_enqueue"
-    }
-  })
-
-  def perform(workflow_id, sms_ids, user_id=nil, subject_set_id=nil)
-    workflow = Workflow.find(workflow_id)
-    user = User.find(user_id) if user_id
-    unless sms_ids.blank?
-      SubjectQueue.dequeue(workflow, sms_ids, user: user, set_id: subject_set_id)
-    end
+  def perform(queue_id, sms_ids)
```

Comment:
Same as enqueue worker, but in this case `workflow_id`s in the queue will get interpreted as `queue_id`s (of which we have more than workflows, so high chance there's a queue for that workflow_id), and Sidekiq might go about removing those sms_ids from that queue.